pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -1700,7 +1700,8 @@ func checkNodeAffinity(og *operationGenerator, volumeToMount VolumeToMount) erro
 		if err != nil {
 			return err
 		}
-		err = util.CheckNodeAffinity(pv, nodeLabels)
+		// TODO(@damemi): As of k/k#95531, `MatchNodeSelectorTerms` called by `CheckNodeAffinity` accepts a *v1.Node instead of a map of nodeLabels
```

Comment:
Phrase TODO as what needs to be done, not just what's the current problem. In this case, perhaps "refactor VolumeHost to return the Node object to pass to CheckNodeAffinity"?

It should be possible:
https://github.com/kubernetes/kubernetes/blob/971477d9b5cc4bf5ae62abe3bbc46e534f481e1b/pkg/kubelet/volume_host.go#L265-L271

But leaving for follow up is preferred.

Reply:
Yeah, I saw that other usage in `pkg/kubelet` where they called `kubelet.GetNode()` but it doesn't look like this implementation of `volumePluginMgr` has any fields that provide a similar function. Updated the TODO for storage folks

Reply:
No need to look at the `volumePluginMgr`, but its `Host`

Reply:
Yeah, neither provides `GetNode` so this function just isn't implemented for storage yet. Though they could probably add it similarly to how kubelet did if they do see value in that!

Reply:
Though, it does have `GetKubeClient()` and `GetNodeName()`, which I could combine to get the node object. Seems unnecessary to add the extra API call though when we already have the labels

Reply:
Can you open a issue to track followup on this?

Reply:
Yup, issue opened here: https://github.com/kubernetes/kubernetes/issues/95837

Reply:
I don't think we should merge passing in a synthetic node... especially one which doesn't resemble a real node (no name, etc)

Reply:
@liggitt the current implementation isn't aware of anything besides the node's labels and fields, so the risk here would only be if `MatchNodeSelector` is ever updated to try to access that info

Regardless, you're right that a synthetic node isn't ideal but the refactoring that would be necessary for a node object to be accessible here should be left to sig-storage. (I'm not very familiar with that code so trying to update it feels like overstepping my bounds here 🙂)

Wdyt about keeping the current wrapper they are using for `CheckNodeAffinity` and just passing the synthetic node there (see changes in https://github.com/kubernetes/kubernetes/pull/95871/files#diff-c1ab67192868d0b9109a57d24f1aab12a796eea2de4f43e4ec1ffd60abe10345R167 for what I mean)? This would minimize the changes we need to make to storage code now, allow the above tracking issue to reference a single point in the code and let them prioritize and make those updates as necessary (ping @msau42 for input too)